secv_guis/bimask_app/main_window.py

```diff
@@ -287,23 +207,31 @@ def new_image(self, img_path, initial_mask_color=(219, 54, 148, 150),
         self.saved_state_tracker = SavedStateTracker()
         return True
 
-    def preannot_from_path(self, preannot_path, rgba, keep_p_value=0.05,
-                           discard_p_value=0.5, normalize=False):
+    def preannot_from_path(self, preannot_path, rgba, upper_thresh=100,
+                           lower_thresh=90, normalize=False):
         """
         This method is prototype-ish: It loads an ``.npz`` file with and
         'entropy' field, expected to have a numpy float matrix with same
-        shape as the image.
+        shape as the image. Alternatively it takes a greyscale image file 
+        suppoted by PIL.
         """
         assert self.scene().img_pmi is not None, \
             "You need to load an image first!"
-        im = Image.open(preannot_path)
-        self._preannot_pmap = np.asarray(im)
+        if preannot_path.endswith(".npz") or preannot_path.endswith(".npy"):
+            self._preannot_pmap = np.load(preannot_path)["entropy"]
+        else:
+            img=np.asanyarray(Image.open(preannot_path))
+            if len(img.shape)>2:
+                img=img[:,:,0]
+            self._preannot_pmap = np.asarray(img)
+            normalize=True
+
         if normalize:
             try:
-                self._preannot_pmap /= self._preannot_pmap.max()
+                self._preannot_pmap = self._preannot_pmap/np.max(self._preannot_pmap)
```

Comment:
how is this better now? in-place was faster and more memory-efficient. Were you having any issues?

Reply:
I got an error because self._preannot_pmap could not be processed this way